app/services/delivery_request_service.rb

```diff
@@ -31,13 +31,21 @@ def call
 
     status = Metrics.email_send_request(provider_name) { send_email }
 
-    return attempt if status == :sent
+    if status == :sent
+      email.update!(sent_at: Time.zone.now, status: :sent)
+      return attempt
+    end
 
     ActiveRecord::Base.transaction do
       attempt.update!(status: status, completed_at: Time.zone.now)
       Metrics.delivery_attempt_status_changed(status)
-      email.mark_as_sent(attempt.finished_sending_at) if attempt.delivered?
-      email.mark_as_failed(attempt.finished_sending_at) if attempt.undeliverable_failure?
+      sent_at = attempt.finished_sending_at
+
+      if attempt.delivered?
+        email.update!(status: :sent, finished_sending_at: sent_at, sent_at: sent_at)
```

Comment:
```suggestion
        email.update!(status: :sent, finished_sending_at: sent_at, sent_at: Time.zone.now)
```

We can be consistent with the line above here